modules/editor/grammar/post.pegjs

```diff
@@ -0,0 +1,95 @@
+Document
+  = WP_Block_List
+
+WP_Block_List
+  = WP_Block*
+
+WP_Block
+  = WP_Block_Balanced
+  / WP_Block_Html
+
+WP_Block_Balanced
+  = s:WP_Block_Start ts:(!WP_Block_End c:Any { return c })+ e:WP_Block_End
+  { return {
+    blockType: s.blockType,
+    attrs: s.attrs,
+    rawContent: ts.join( '' ),
+  } }
+
+WP_Block_Html
+  = ts:(!WP_Block_Balanced c:Any { return c })+
+  {
+    return {
+      blockType: 'html',
+      attrs: {},
+      rawContent: ts.join('')
+    }
+  }
+
+WP_Block_Start
+  = "<!--" __ "wp:" blockType:WP_Block_Type attrs:WP_Block_Attribute_List _? "-->"
+  { return {
+    type: 'WP_Block_Start',
+    blockType,
+    attrs,
+    text: text()
+  } }
+
+WP_Block_End
+  = "<!--" __ "/wp" __ "-->"
```

Comment:
I'm ok making this change if we reach a consensus. I feel that it hasn't been discussed yet.

I'm personally neutral, I don't think it adds much value, (I see the HTML as a serialisation that should not be updated frequently by hand, it's our job to make the best possible editor to avoid having to use the HTML editor) but I understand that some might find it more readable.

@aduth @mtias thoughts